src/db/sqldb/history_sqldb_test.go

```diff
@@ -0,0 +1,347 @@
+				Expect(err).NotTo(HaveOccurred())
+				Expect(hasScalingHistory("an-app-id", 111111)).To(BeTrue())
+			})
+		})
+
+		Context("When inserting multiple scaling history records of an app", func() {
+			BeforeEach(func() {
+				history = &models.AppScalingHistory{
+					ScalingType:  models.ScalingTypeSchedule,
+					Status:       models.ScalingStatusFailed,
+					OldInstances: 2,
+					NewInstances: 4,
+					Reason:       "a reason",
+					Message:      "a message",
+					Error:        "an error",
+				}
+				history.AppId = "an-app-id"
+				history.Timestamp = 111111
+				err = hdb.SaveScalingHistory(history)
+				Expect(err).NotTo(HaveOccurred())
+
+				history.AppId = "an-app-id"
+				history.Timestamp = 222222
+				err = hdb.SaveScalingHistory(history)
+				Expect(err).NotTo(HaveOccurred())
+
+				history.AppId = "another-app-id"
+				history.Timestamp = 333333
+				err = hdb.SaveScalingHistory(history)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			It("has all the histories in database", func() {
+
+				Expect(hasScalingHistory("an-app-id", 111111)).To(BeTrue())
+				Expect(hasScalingHistory("an-app-id", 222222)).To(BeTrue())
+				Expect(hasScalingHistory("another-app-id", 333333)).To(BeTrue())
+
+			})
+		})
+
+	})
+
+	Describe("RetrieveScalingHistories", func() {
+
+		BeforeEach(func() {
+			hdb, err = NewHistorySQLDB(url, logger)
+			Expect(err).NotTo(HaveOccurred())
+			cleanScalingHistoryTable()
+
+			start = 0
+			end = -1
+			appId = "an-app-id"
+		})
+
+		AfterEach(func() {
+			err = hdb.Close()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			history = &models.AppScalingHistory{
+				AppId:        appId,
+				OldInstances: 2,
+				NewInstances: 4,
+				Reason:       "a reason",
+				Message:      "a message",
+			}
+
+			history.Timestamp = 666666
+			history.ScalingType = models.ScalingTypeDynamic
+			history.Status = models.ScalingStatusSucceeded
+			history.Error = ""
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 222222
+			history.ScalingType = models.ScalingTypeDynamic
+			history.Status = models.ScalingStatusFailed
+			history.Error = "an error"
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 555555
+			history.ScalingType = models.ScalingTypeSchedule
+			history.Status = models.ScalingStatusFailed
+			history.Error = "an error"
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			history.Timestamp = 333333
+			history.ScalingType = models.ScalingTypeSchedule
+			history.Status = models.ScalingStatusIgnored
+			history.Error = ""
+			err = hdb.SaveScalingHistory(history)
+			Expect(err).NotTo(HaveOccurred())
+
+			histories, err = hdb.RetrieveScalingHistories("an-app-id", start, end)
+		})
+
+		Context("When the app has no hisotry", func() {
+			BeforeEach(func() {
+				appId = "other-app-id"
```

Comment:
I would do this the reverse.
Have a fixed Id for the data in the DB, and use this for the retrieval key. It makes the testing a bit more consistent.